cpp/src/aztec/honk/composer/composer_helper/standard_plonk_composer_helper.cpp

```diff
@@ -0,0 +1,205 @@
+#include "standard_plonk_composer_helper.hpp"
+#include <polynomials/polynomial.hpp>
+#include <proof_system/flavor/flavor.hpp>
+#include <honk/pcs/commitment_key.hpp>
+#include <numeric/bitop/get_msb.hpp>
+#include <plonk/proof_system/widgets/transition_widgets/arithmetic_widget.hpp>
+#include <plonk/proof_system/widgets/random_widgets/permutation_widget.hpp>
+#include <plonk/proof_system/commitment_scheme/kate_commitment_scheme.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+
+namespace bonk {
+
+/**
+ * Compute proving key base.
+ *
+ * 1. Load crs.
+ * 2. Initialize this.circuit_proving_key.
+ * 3. Create constraint selector polynomials from each of this composer's `selectors` vectors and add them to the
+ * proving key.
+ *
+ * N.B. Need to add the fix for coefficients
+ *
+ * @param minimum_circuit_size Used as the total number of gates when larger than n + count of public inputs.
+ * @param num_reserved_gates The number of reserved gates.
+ * @return Pointer to the initialized proving key updated with selector polynomials.
+ * */
+template <typename CircuitConstructor>
+std::shared_ptr<bonk::proving_key> StandardPlonkComposerHelper<CircuitConstructor>::compute_proving_key_base(
+    const CircuitConstructor& constructor, const size_t minimum_circuit_size, const size_t num_randomized_gates)
+{
+
+    // Initialize circuit_proving_key
+    // TODO: replace composer types.
+    circuit_proving_key = initialize_proving_key(
+        constructor, crs_factory_.get(), minimum_circuit_size, num_randomized_gates, plonk::ComposerType::STANDARD);
+    // Compute lagrange selectors
+    put_selectors_in_polynomial_cache(constructor, circuit_proving_key.get());
```

Comment:
As I alluded to earlier, I think I would prefer e.g. `compute_selectors_lagrange()` and `compute_selectors_monomial_and_coset_fft()` here. Then we could have comments in this function (and the sub-functions) saying that they are also added to the key. Otherwise it's misleading to not include "add_to_cache" in other functions like `compute_witness_base`.